### PR TITLE
adding tcp connection support

### PR DESCRIPTION
**Pull request by jkretzmer**
Adding support for connection to receiver over TCP.
Adjusting print statements to be more generic, referencing "Swift receiver" rather than "Piksi" since the driver also supports the Duro receiver.

### CODE REVIEW

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
+# interface: specifies the interface over which to connect.
+# supported values are 'serial' and 'tcp'
+# if this value is not valid, the driver will default to serial communication
+interface: 'serial'
+
 base_station_mode: false
-debug_mode: false # if true, output every topic read by this driver
+
+# debug_mode: if true, output every topic read by this driver
 # whenever possible, use network broadcast, not global ip broadcast
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: The comment " whenever possible, use ..." geos before to parameter "broadcast_addr" and not before param "debug_mode".

File: piksi_multi_rtk_ros/install/install_piksi_multi.sh [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 GIT_REPO_LIBSBP=https://github.com/swift-nav/libsbp.git
-REPO_TAG=v2.2.15 #version you want to checkout before installing
+REPO_TAG=v2.3.11 #version you want to checkout before installing
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/install/install_piksi_multi.sh]: Why do you want to change this in master? Master has been tested with SBP 2.2.15, as written here https://github.com/ethz-asl/ethz_piksi_ros/tree/master/piksi_multi_rtk_ros#firmware-and-sbp-lib-version

I'm working on another pull request to switch to 2.3.10: https://github.com/ethz-asl/ethz_piksi_ros/pull/16

File: piksi_multi_rtk_ros/src/piksi_multi.py [hunk 3]
```diff
@@ -45,7 +47,7 @@
 
 
 class PiksiMulti:
-    LIB_SBP_VERSION_MULTI = '2.2.15'  # SBP version used for Piksi Multi.
+    LIB_SBP_VERSION_MULTI = '2.3.11'  # SBP version used for Piksi Multi.
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/src/piksi_multi.py]: Same here, please leave 2.2.15

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml [hunk 4]
```diff
@@ -1,20 +1,56 @@
...
 var_rtk_fix: [0.0049, 0.0049, 0.01]
+
+# Serial Settings
+
+# serial_port: defines which serial device to use for communication
+# ex '/dev/ttyUSB0'
+serial_port: '/dev/ttyUSB0'
+
+# baud_rate: defines the rate for the serial connection. This driver uses
+# 234000 as the default. Swift Navigation receivers ship with a
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: Typo? Correct value is `230400`

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml [hunk 5]
```diff
@@ -1,20 +1,56 @@
...
+# default baud rate of 115200.
+baud_rate: 234000
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: Typo? Correct value is `230400`

File: piksi_multi_rtk_ros/src/piksi_multi.py [hunk 6]
```diff
@@ -75,28 +77,38 @@ def __init__(self):
...
+            tcp_addr = rospy.get_param('~tcp_addr', '192.168.0.222')
+            tcp_port = rospy.get_param('~tcp_port', 55555)
+            try:
+                self.driver = TCPDriver(tcp_addr, tcp_port)
+            except SystemExit:
+                rospy.logerr("Unable to open TCP connection %s:%s", (tcp_addr, tcp_port))
+                raise
+        else:
+            serial_port = rospy.get_param('~serial_port', '/dev/ttyUSB0')
+            baud_rate = rospy.get_param('~baud_rate', 23400)
```
Comment (marco-tranzatto) [inline piksi_multi_rtk_ros/src/piksi_multi.py]: Typo? Correct value is `230400`

File: piksi_multi_rtk_ros/install/install_piksi_multi.sh (hunk 2 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/install/install_piksi_multi.sh]: I had been working against 2.3.11 (which was the newest tag of SBP at the time) but 2.3.10 would be fine as well. I will adjust the PR.

File: piksi_multi_rtk_ros/src/piksi_multi.py (hunk 3 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/src/piksi_multi.py]: Same as previous.

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml (hunk 4 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: good catch - will fix.

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml (hunk 5 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: good catch - will fix.

File: piksi_multi_rtk_ros/src/piksi_multi.py (hunk 6 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/src/piksi_multi.py]: good catch - will fix.

File: piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml (hunk 1 above)
Comment (jkretzmer) [reply piksi_multi_rtk_ros/cfg/piksi_multi_driver_settings.yaml]: good catch. will update.

File: piksi_multi_rtk_ros/install/install_piksi_multi.sh (hunk 2 above)
Comment (marco-tranzatto) [reply piksi_multi_rtk_ros/install/install_piksi_multi.sh]: We just deiced to stick to the indication here: [Piksi Multi and Duro SBP Specifications](https://support.swiftnav.com/customer/en/portal/articles/2492810-swift-binary-protocol).

I'd like to test my PR today and merge it asap ;)

**Pull request merged**